server/internal/core/application/covenantless.go

```diff
@@ -1080,16 +1078,24 @@ func (s *covenantlessService) startFinalization() {
 	s.forfeitTxs.init(connectors, requests)
 
 	if len(vtxoTree) > 0 {
-		log.Debugf("signing vtxo tree for round %s", round.Id)
 
-		signingSession := newMusigSigningSession(len(cosigners))
+		uniqueSignerPubkeys := make(map[string]struct{})
```

Comment:
if we added the server key only after popping the requests, we could just count the number of cosigners instead of using this map

Reply:
No, cause you don't know if 2 different TxRequest are using the same cosigner